### PR TITLE
Ignore warnings from ldoc by default

### PR DESCRIPTION
**Pull request by psychon**
A while ago, we made errors from ldoc fatal by default. Then a new ldoc
release appeared and caused problems for all of our users, because
awesome failed to work.

This patch reverts the previous fix so that we ignore ldoc warnings by
default again. However, to catch ldoc warnings on Travis, another
ldoc-building-target is added that fails on warnings. This new target is
included in our "check" target.

Helps-with: https://github.com/awesomeWM/awesome/issues/1098
Signed-off-by: Uli Schlachter <psychon@znc.in>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -272,17 +274,24 @@ if(GENERATE_DOC)
     # Copy the aliases to the build directory
     file(COPY ${SOURCE_DIR}/docs/aliases DESTINATION ${BUILD_DIR}/docs)
 
+    add_custom_target(ldoc ALL
+        COMMAND ${LDOC_EXECUTABLE} .
+        WORKING_DIRECTORY ${AWE_DOC_DIR}
+        DEPENDS ${AWE_LUA_FILES} ${AWE_MD_FILES} ${BUILD_DIR}/docs/06-appearance.md
+        ${BUILD_DIR}/docs/05-awesomerc.md
+    )
```
Comment (blueyed) [inline CMakeLists.txt]: I think we should use `.../docs/*.md.lua` (with the `.lua` extension removed) in the DEPENDS here.

**Comment by Elv13** [comment]
Please also add `Fix #1098` in the commit message. While it doesn't "blacklist" anything, it does solve the intent of the issue (users not being able to install awesome).

**Comment by actionless** [comment]
this solves the build issue for me

File: CMakeLists.txt (hunk 1 above)
Comment (blueyed) [inline CMakeLists.txt]: @psychon 
Can you address this please, if it makes sense?

Comment (psychon) [inline CMakeLists.txt]: (1) CMake is ugly. (2) Done.

**Comment by psychon** [comment]
> Please also add Fix #1098 in the commit message.

Done (although I still prefer full URLs instead of this short form; we already have lots of "FS#1234"-references in the git history that are now that we do not use Flyspray anymore a lot harder to understand; but I seem to be alone with this opinion)

**Comment by blueyed** [comment]
> but I seem to be alone with this opinion

No, I also prefer the long form.

Thanks!

**Pull request merged**